File: Codility/TimeComplexity/PermMissingElem.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")


def solution(A):
    # write your code in Python 3.6
    n = len(A)
    total = (n + 2) * (n + 1) / 2
    for i in A:
        total -= i
    return int(total)


def solution(A):
    # write your code in Python 3.6
    ret = len(A)

    for i in range(len(A)):
        ret = ret ^ i ^ (A[i] - 1)

    return ret + 1

File: Codility/TimeComplexity/PermMissingElem.cpp
// you can use includes, for example:
// #include <algorithm>
#include <algorithm>
// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(vector<int> &A) {
    // write your code in C++14 (g++ 6.2.0)

    // option 1 
    // int n = A.size();
    // int total = (n+2)*(n+1)/2;
    // for(int a:A) total -= a;
    // return total;

    // option 2
    int n = A.size();
    int ret = n;
    for(int i=0;i<n;++i){
        ret ^= i^(A[i]-1);
    }
    return ret+1;


}